helix-term/src/ui/editor.rs

```diff
@@ -227,16 +227,14 @@ impl EditorView {
         _theme: &Theme,
     ) -> Box<dyn Iterator<Item = HighlightEvent> + 'doc> {
         let text = doc.text().slice(..);
-        let last_line = std::cmp::min(
-            // Saturating subs to make it inclusive zero indexing.
-            (offset.row + height as usize).saturating_sub(1),
-            doc.text().len_lines().saturating_sub(1),
-        );
+        // Saturating subs to make it inclusive zero indexing.
+        let last_line = doc.text().len_lines().saturating_sub(1);
+        let view_end_line = (offset.row + height as usize).saturating_sub(1);
```

Comment:
I'd move both of these into the `let range` block, and rename `let last_visible_line = (offset.row + height as usize).saturating_sub(1).min(last_line)`